pkg/cmd/pipelineresource/create.go

```diff
@@ -0,0 +1,588 @@
+			return err
+		}
+	}
+
+	cls, err := res.params.Clients()
+	if err != nil {
+		return err
+	}
+
+	newRes, err := cls.Tekton.TektonV1alpha1().PipelineResources(res.params.Namespace()).Create(&res.pipelineResource)
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintf(res.stream.Out, "New %s resource \"%s\" created\n", newRes.Spec.Type, newRes.Name)
+	return nil
+}
+
+func (res *resource) askMeta() error {
+	var answer string
+	var qs = []*survey.Question{{
+		Name: "resource name",
+		Prompt: &survey.Input{
+			Message: "Name of the pipeline resource :",
+		},
+		Validate: survey.Required,
+	}}
+
+	err := survey.Ask(qs, &answer, res.askOpts)
+	if err != nil {
+		return Error(err)
+	}
+	if err := validate(answer, res.params); err != nil {
+		return err
+	}
+
+	res.pipelineResource.Name = answer
+
+	return nil
+}
+
+func (res *resource) askType() error {
+	var answer string
+	var qs = []*survey.Question{{
+		Name: "pipelineResource",
+		Prompt: &survey.Select{
+			Message: "Select resource type to create :",
+			Options: allResourceType(),
+		},
+	}}
+
+	err := survey.Ask(qs, &answer, res.askOpts)
+	if err != nil {
+		return Error(err)
+	}
+
+	res.pipelineResource.Spec.Type = cast(answer)
+
+	return nil
+}
+
+func (res *resource) askGitParams() error {
+	urlParam, err := askParam("url", res.askOpts)
+	if err != nil {
+		return err
+	}
+	if urlParam.Name != "" {
+		res.pipelineResource.Spec.Params = append(res.pipelineResource.Spec.Params, urlParam)
+	}
+
+	revisionParam, err := askParam("revision", res.askOpts)
+	if err != nil {
+		return err
+	}
+	if revisionParam.Name != "" {
+		res.pipelineResource.Spec.Params = append(res.pipelineResource.Spec.Params, revisionParam)
+	}
+
+	return nil
+}
+
+func (res *resource) askStorageParams() error {
+	options := []string{"gcs", "build-gcs"}
+
+	storageType, err := askToSelect("Select storage type", options, res.askOpts)
+	if err != nil {
+		return err
+	}
+	param := v1alpha1.ResourceParam{}
+	param.Name, param.Value = "type", storageType
+	res.pipelineResource.Spec.Params = append(res.pipelineResource.Spec.Params, param)
+
+	switch storageType {
+	case options[0]: // for storage type as gcs
+		locationParam, err := askParam("location", res.askOpts)
+		if err != nil {
+			return err
+		}
+		if locationParam.Name != "" {
+			res.pipelineResource.Spec.Params = append(res.pipelineResource.Spec.Params, locationParam)
+		}
+
+		dirParam, err := askParam("dir", res.askOpts)
+		if err != nil {
+			return err
+		}
+		if dirParam.Name != "" {
+			res.pipelineResource.Spec.Params = append(res.pipelineResource.Spec.Params, dirParam)
+		}
+
+	case options[1]: // for storage type as build-gcs
+		locationParam, err := askParam("location", res.askOpts)
+		if err != nil {
+			return err
+		}
+		if locationParam.Name != "" {
+			res.pipelineResource.Spec.Params = append(res.pipelineResource.Spec.Params, locationParam)
+		}
+
+		artifactTypeParam, err := askParam("artifactType", res.askOpts)
```

Comment:
can we have plain english instead of just artifcatType?

same for others,

Reply:
Yes, this also need to be change as a select question artifactType could be only zipArchive or tarArchive or Manifest.
Thanks for pointing this.